internal/pipe/ko/ko.go

```diff
@@ -0,0 +1,269 @@
+type Pipe struct{}
+
+func (Pipe) String() string { return "running ko" }
+func (Pipe) Skip(ctx *gcontext.Context) bool {
+	return ctx.SkipKo || ctx.Config.Ko.ID == ""
+}
+
+// Default sets the Pipes defaults.
+func (Pipe) Default(ctx *gcontext.Context) error {
+	ids := ids.New("kos")
+	cfg := &ctx.Config.Ko
+	if err := setConfigDefaults(cfg); err != nil {
+		return err
+	}
+	ids.Inc(cfg.ID)
+	return ids.Validate()
+}
+
+func setConfigDefaults(cfg *config.Ko) error {
+	cfg.Push = true
+
+	if cfg.ID == "" {
+		cfg.ID = "default"
+	}
+
+	if cfg.BaseImage == "" {
+		cfg.BaseImage = "cgr.dev/chainguard/static" // TODO: we can discuss on this
+	}
+
+	return nil
+}
+
+// Run executes the Pipe.
+func (Pipe) Run(ctx *gcontext.Context) error {
+	g := semerrgroup.New(ctx.Parallelism)
+	g.Go(doBuild(ctx))
+	return g.Wait()
+}
+
+type buildOptions struct {
+	ip                   string
+	workingDir           string
+	dockerRepo           string
+	cosignRepo           string
+	platforms            []string
+	baseImage            string
+	tags                 []string
+	sbom                 string
+	ldflags              []string
+	pkg                  string
+	bare                 bool
+	preserverImportPaths bool
+	baseImportPaths      bool
+}
+
+var baseImages sync.Map
+
+func (o *buildOptions) makeBuilder(ctx context.Context) (*build.Caching, error) {
+	bo := []build.Option{
+		build.WithConfig(map[string]build.Config{
+			o.pkg: {
+				Ldflags: o.ldflags,
+			},
+		}),
+		build.WithPlatforms(o.platforms...),
+		build.WithBaseImages(func(ctx context.Context, s string) (name.Reference, build.Result, error) {
+			ref, err := name.ParseReference(o.baseImage)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			if cached, found := baseImages.Load(o.baseImage); found {
+				return ref, cached.(build.Result), nil
+			}
+
+			desc, err := remote.Get(ref,
+				remote.WithAuthFromKeychain(authn.DefaultKeychain))
+			if err != nil {
+				return nil, nil, err
+			}
+			if desc.MediaType.IsImage() {
+				img, err := desc.Image()
+				baseImages.Store(o.baseImage, img)
+				return ref, img, err
+			}
+			if desc.MediaType.IsIndex() {
+				idx, err := desc.ImageIndex()
+				baseImages.Store(o.baseImage, idx)
+				return ref, idx, err
+			}
+			return nil, nil, fmt.Errorf("unexpected base image media type: %s", desc.MediaType)
+		}),
+	}
+	switch o.sbom {
+	case "spdx":
+		bo = append(bo, build.WithSPDX("devel"))
+	case "cyclonedx":
+		bo = append(bo, build.WithCycloneDX())
+	case "go.version-m":
+		bo = append(bo, build.WithGoVersionSBOM())
+	case "none":
+		// don't do anything.
+	default:
+		return nil, fmt.Errorf("unknown sbom type: %q", o.sbom)
+	}
+
+	b, err := build.NewGo(ctx, o.workingDir, bo...)
+	if err != nil {
+		return nil, fmt.Errorf("NewGo: %v", err)
+	}
+	return build.NewCaching(b)
+}
+
+func doBuild(ctx *gcontext.Context) func() error {
+	return func() error {
+		ctxBackground, cancel := context.WithCancel(context.Background())
+		defer cancel()
```

Comment:
I believe we can use the current goreleaser context, no?

Reply:
I tried to use the goreleaser one, but it didn't work, so I switched it with the standard one.

Reply:
hmm weird, will check later